tracer/src/Datadog.Trace/Debugger/Snapshots/DebuggerSnapshotCreator.cs

```diff
@@ -68,15 +69,9 @@ internal CaptureBehaviour CaptureBehaviour
             }
         }
 
-        public static DebuggerSnapshotCreator BuildSnapshotCreator(string probeId)
+        public static DebuggerSnapshotCreator BuildSnapshotCreator(ProbeProcessor processor)
         {
-            var probeInfo = ProbeExpressionsProcessor.Instance.GetProbeInfo(probeId);
-            if (probeInfo.HasValue)
-            {
-                return new DebuggerSnapshotCreator(probeInfo.Value.IsFullSnapshot, probeInfo.Value.ProbeLocation, probeInfo.Value.HasCondition);
-            }
-
-            throw new InvalidOperationException("Probe info not found for probe id: " + probeId);
+            return new DebuggerSnapshotCreator(processor.ProbeInfo.IsFullSnapshot, processor.ProbeInfo.ProbeLocation, processor.ProbeInfo.HasCondition);
```

Comment:
If probe is updated, do we creating a new `DebuggerSnapshotCreator`? it's not related only to this PR but as general, because we save a readonly data in the snapshot creator.

Reply:
`DebuggerSnapshotCreator` is one instance per probe execution, so every probe execution is having a new instance.
In regard to template/condition editing - they shall work the same because we still have the `ConcurrentDictionary` that maps `ProbeId` -> `ProbeProcessor`, so even though a probe is still "executing" it's based on heap-allocated `ProbeProcessor`, so the same instance will be impacted thus even executing probes should receive these updates.